tests/test_integration_classification_2d.py

```diff
@@ -240,6 +240,33 @@ def test_training(self):
 
         np.testing.assert_allclose(repeated[0], repeated[1])
 
+    @TimedCall(seconds=400, skip_timing=not torch.cuda.is_available(), daemon=False)
+    def test_timing(self):
```

Comment:
I think most of the logic in this function is duplicated to `test_training`, could you please combine them or extract a common internal function?